src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1079,7 +1078,15 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                             // initializers for global code have already been included in the body
                             if (includeInitializersInBody)
                             {
-                                boundStatements = boundStatements.Concat(processedInitializers.LoweredInitializers.Statements);
+                                if (processedInitializers.LoweredInitializers.Kind == BoundKind.StatementList)
+                                {
+                                    BoundStatementList lowered = (BoundStatementList) processedInitializers.LoweredInitializers;
+                                    boundStatements = boundStatements.Concat(lowered.Statements);
+                                }
+                                else
+                                {
+                                    boundStatements = boundStatements.Add(processedInitializers.LoweredInitializers);
```

Comment:
@gafter Maybe.  I'm not sure if that will change the way that we deal with these statements later on though.